### PR TITLE
[VIDEO-2688] - Install Windows MSIs from custom source

### PR DESCRIPTION
**Pull request by lyramcmillan**
## Description

[//]: # (Add a brief, one or two line description of the changes below)
Updates supporting downloading Windows MSIs from an overridden URI with uninstall clean up. Added optional inputs and code to install MSIs from specified URLs using optional build run parameter for isolating the install location. Also has post build action cleanup uninstall of gstream,er

**Ticket(s) Addressed:**

[//]: # (Add all tickets addressed or related to this PR with links to JIRA)
* VIDEO-2688

## Design Summary

[//]: # (Add a brief summary of the design of the changes as appropriate below, add a link to confluence if applicable)

**Configuration changes:**

[//]: # (Define any changes to the settings or other flags, be sure to update the wiki page)

## Testing Notes and Procedure

[//]: # (Add a description and steps for testing the changes)

## Related Pull Requests

[//]: # (Add links to all related pull requests if applicable, edit this comment to add links when other requests are opened)
* blinemedical/project#xxx

### CODE REVIEW

**Comment by saraboule** [comment]
@lyramcmillan is this ready for review? looks like some unit tests are failing.

**Comment by lyramcmillan** [comment]
> @lyramcmillan is this ready for review? looks like some unit tests are failing.

Nope - trying to figure out why tests are failing. There's something bad in the test setup I think but it's proving intractable. Making into a draft.

File: README.md [hunk 1]
```diff
@@ -22,6 +22,21 @@ The architecture (`x86`, `x86_64`, etc.) of binaries to install.  Defaults to `x
...
+**WINDOWS ONLY:**
+
+### `msiUrl`
+
+An optional override url for downloading the Windows gstreamer runtime MSI installer. If not specified, the default is `https://gstreamer.freedesktop.org/data/pkg/windows/<version>/gstreamer-1.0-<arch>-<version>.msi`. If this input is specified then the action post run will be run and gsrtreamer will be uninstalled. These are primarily meant to be used in self hosted runners and not run in the gstreamer hosted runners that are bootstrapped and executed as non-persistent containers.
+
+### `devMsiUrl`
+
+An optional override url for downloading the Windows gstreamer development libraries and tools MSI installer. If not specified, the default is `https://gstreamer.freedesktop.org/data/pkg/windows/<version>/gstreamer-1.0-devel-<arch>-<version>.msi`.
+If the msiUrl is specified the devMsiUrl must also be specified.
```
Comment (saraboule) [inline README.md:34]: techinically a run-on sentence. just add a comma there, or was that supposed to be then?

File: README.md [hunk 2]
```diff
@@ -22,6 +22,21 @@ The architecture (`x86`, `x86_64`, etc.) of binaries to install.  Defaults to `x
 
 The URL from where to clone the gstreamer source (Linux targets only).  Defaults to `https://gitlab.freedesktop.org/gstreamer/gstreamer.git`.
 
+**WINDOWS ONLY:**
+
+### `msiUrl`
+
+An optional override url for downloading the Windows gstreamer runtime MSI installer. If not specified, the default is `https://gstreamer.freedesktop.org/data/pkg/windows/<version>/gstreamer-1.0-<arch>-<version>.msi`. If this input is specified then the action post run will be run and gsrtreamer will be uninstalled. These are primarily meant to be used in self hosted runners and not run in the gstreamer hosted runners that are bootstrapped and executed as non-persistent containers.
```
Comment (saraboule) [inline README.md:29]: `gsrtreamer`

**Review by saraboule — APPROVED** [review]
will need a follow up on why the windows-2022 unit tests are failing. otherwise lets go!!!

**Pull request merged**